src/platform.ts

```diff
@@ -80,14 +80,16 @@ export class MideaPlatform implements DynamicPlatformPlugin {
   private finishedLaunching() {
     this.log.info('Start device discovery...');
     // If IP address is in config then probe them directly
-    Object.values(this.platformConfig.devices).forEach((device) => {
-      // for some reason, assigning the regex has to be inside the loop, else fails after first pass.
-      const regexIPv4 = /^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])$/gi;
-      const ip = device.advanced_options?.ip?.toString().trim();
-      if (regexIPv4.test(ip)) {
-        this.discover.discoverDeviceByIP(ip);
-      } else if (ip) {
-        this.log.warn(`[${device.name}] Invalid IP address in configuration: ${ip}`);
+    this.platformConfig.devices.forEach((device) => {
+      if (device.advanced_options.ip.length > 0) {
```

Comment:
We don't need to test for zero length, regex will fail.  And we need to handle undefined advanced_options and ip

Reply:
They can't be undefined as they have default values.

Reply:
> They can't be undefined as they have default values.

In theory.  But I can tell you it crashed for me during testing (my version, when I was working on it).  Remember that for the Homebridge config we have stripped out settings that are unchanged from default.

Reply:
The length check is needed, because the regex will fail for every device which has the default IP value (an empty string), thus logging it out.

Reply:
> The length check is needed, because the regex will fail for every device which has the default IP value (an empty string), thus logging it out.

No, I'm wrong here, you have added the test of the ip varible in the else case. I think my solution is much cleaner: why would we test for regex when we know that the string is an empty string?

Reply:
Yes the regex will fail, so no need to do discovery by IP.  But it is not logged because I test for non-empty.

Reply:
> > The length check is needed, because the regex will fail for every device which has the default IP value (an empty string), thus logging it out.
> 
> No, I'm wrong here, you have added the test of the ip varible in the else case. I think my solution is much cleaner: why would we test for regex when we know that the string is an empty string?

It was an extra sanity check.  But I'm fine with removing the regex and just accepting anything in the entry field and letting an error throw somewhere else if that is your preference.

Reply:
> We don't need to test for zero length, regex will fail. And we need to handle undefined advanced_options and ip

Yeah you are right, I merge the default config to the device config later in the code.

Reply:
I'm okay with the regex, I'm just saying I wouldn't test for it, when the ip is empty.